### PR TITLE
Fix ConcurrentModificationException #894

### PR DESCRIPTION
**Pull request by kevind-wgu**
See Issue #894 for an explination.

### CODE REVIEW

**Comment by altro3** [comment]
@graemerocher Hi! Could you merge these fixes: this, #892, #891, #888 ? And finalize release 4.8.2.

File: openapi/src/main/java/io/micronaut/openapi/visitor/OpenApiApplicationVisitor.java [hunk 1]
```diff
@@ -942,7 +943,9 @@ public void finish(VisitorContext visitorContext) {
                 Map<String, Schema> schemas = openAPI.getComponents().getSchemas();
                 if (CollectionUtils.isNotEmpty(schemas)) {
                     String openApiJson = ConvertUtils.getJsonMapper().writeValueAsString(openAPI);
-                    for (String schemaName : schemas.keySet()) {
+                    // Create a copy of the keySet so that we can modify the map while in a foreach
+                    Set<String> keySet = schemas.keySet().stream().collect(Collectors.toSet());
```
Comment (altro3) [inline openapi/src/main/java/io/micronaut/openapi/visitor/OpenApiApplicationVisitor.java:947]: don't need to create stream for it. It's not efficient. Just write like this:
```java
Set<String> keySet = new HashSet<>(schemas.keySet());
```

Comment (altro3) [reply openapi/src/main/java/io/micronaut/openapi/visitor/OpenApiApplicationVisitor.java:947]: @kevind-wgu could you fix it?

Comment (kevind-wgu) [reply openapi/src/main/java/io/micronaut/openapi/visitor/OpenApiApplicationVisitor.java:947]: I assumed that using a stream was efficient as directly using a HashSet. Yes I will change it.

Comment (kevind-wgu) [reply openapi/src/main/java/io/micronaut/openapi/visitor/OpenApiApplicationVisitor.java]: @altro3 it is fixed.

**Review by altro3 — APPROVED** [review]

**Comment by altro3** [comment]
@graemerocher ping. Hi! Could you merge these fixes: this, https://github.com/micronaut-projects/micronaut-openapi/pull/892, https://github.com/micronaut-projects/micronaut-openapi/pull/891, https://github.com/micronaut-projects/micronaut-openapi/pull/888 ? And finalize release 4.8.2.

**Pull request merged**